packages/firestore/src/util/misc.ts

```diff
@@ -28,8 +29,17 @@ export class AutoId {
     const chars =
       'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
     let autoId = '';
-    for (let i = 0; i < 20; i++) {
-      autoId += chars.charAt(Math.floor(Math.random() * chars.length));
+    while (autoId.length < 20) {
+      const bytes = PlatformSupport.getPlatform().randomBytes(40);
+      for (let i = 0; i < bytes.length; ++i) {
+        // Length of `chars` is 62. We only take bytes between 0 and 62*4-1
+        // (both inclusive). The value is then evenly mapped to indices of `char`
+        // via a modulo operation.
+        const maxValue = 62 * 4 - 1;
```

Comment:
Hardcoding `62` in three different places (including in the comment) is a bit unnerving.  It would be preferable to use `chars.length`, for robustness. How about something like this:

```js
const chars = '...';
const maxValue = chars.length * 4 - 1;
debugAssert(maxValue < 256);
```

Then the hardcoded value of `62` in the modulo operation below can be safely replaced with `chars.length`.